Refactor

williamputraintan opened the pull request:
- Fix function bug
- More Comments and logging

Commit:
Change Image pull from DockerHub to ECR

williamputraintan committed:
Added missing Function Parameter

williamputraintan committed:
Variable Refactoring

williamputraintan committed:
Update Commenting and Logger

reisingerf commented on lambdas/functions/main.py:
```diff
@@ -2,123 +2,155 @@
+    log_path = LOG_DIRECTORY["samplesheet_check"]
+
     # Check if data input is correct
     if log_level in ["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"] == False:
```
perhaps use `if log_level not in [...]:` for readability?

williamputraintan replied:
Fix on 45b3cf60a7ec457f6e6ada6f01909295e8dcb91a

reisingerf commented on lambdas/functions/main.py:
```diff
@@ -2,123 +2,155 @@
+
+        # Return response
+        body = construct_body(check_status="FAIL", error_message="IncorrectLogLevel", log_path=log_path)
+        response = construct_response(status_code=200, body=body, origin=origin)
```
200 response? Because the request was successful (syntax correct) even thought the payload was unexpected (semantics wrong)?

williamputraintan replied:
Hmm... I think it's still 200 since it's a valid payload just that the content is incorrect? Initially, I put it as 40X but then will it be the same thing if the file we are checking had invalid content but the payload is correct? That happened in the previous version where invalid sample sheet data had incorrect format (missplaced commas) and return 40X status code, and the front end interpret it as a network error which shows the pop-up dialog saying 'an error has occurred'

But, Im happy to change it to a 40X status code if it makes more sense tho

reisingerf replied:
That's fine. It's what I meant with the syntax being correct.

reisingerf commented on stacks/sscheck_backend_stack.py:
```diff
@@ -64,6 +64,7 @@ def __init__(self, scope: cdk.Construct, construct_id: str, **kwargs) -> None:
         sample_sheet_check_lambda = lambda_.Function(
             self,
             "SampleSheetValidationLambda",
+            function_name="sscheck-backend-lambda",
```
Minor comment: This is a name for a lambda, naming it with `-lambda` may be redundant?
(happy to leave it as is though)

williamputraintan replied:
Fix on 45b3cf60a7ec457f6e6ada6f01909295e8dcb91a

reisingerf reviewed (dismissed):
Minor comments, otherwise LGTM

williamputraintan committed:
Minor Refacotring

- Remove unused sscheck script code
- Rename Lambdas
- Refactor code readability

williamputraintan committed:
Merge branch 'main' into dev

victorskl commented on lambdas/functions/main.py:
```diff
@@ -2,123 +2,155 @@
+    logger.info(f'Parsing origin header content: {origin}')
+
     auth_header = event["headers"]["Authorization"]
+    logger.info(f'Parsing auth_header header content: {auth_header}')
```
I'd recommend, not to log auth header content here.

williamputraintan replied:
Removed at commit 3c49e1c

williamputraintan committed:
Remove auth_header logging

victorskl reviewed (approved):
LGTM!
